### ISSUE TITLE
[BUG] Checks fail on GitHub Actions if analyzer version set >= 4.2.0

### ISSUE THREAD

**Issue by lavinov-mercury** [dart-code-checker/dart-code-metrics#982]
This issue occurs for check-unused-code, check-unused-files & check-unused-l10n on GitHub actions when analyzer version set to 4.2 or higher (up to 4.6)

This most likely related to https://github.com/dart-code-checker/dart-code-metrics/issues/811, but i can't be sure. Issues have the same preconditions, but when i excluded files, which caused mentioned issue, CI proceeded to fail (check [comment](https://github.com/dart-code-checker/dart-code-metrics/issues/811#issuecomment-1234032888) for details)
 
Also, is there any way to collect more data on CI? I haven't found any flag like `--verbose` in docs, bit it would be handy in this case. 

* **Dart code metrics version:** 4.17.1
* **Dart sdk version:** 2.17 (& 2.18)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
# enable all rules by default
include: all_lint_rules.yaml

analyzer:
  language:
    strict-casts: true
  errors:
    # otherwise cause the import of all_lint_rules to warn because of some rules conflicts.
    # the conflicts are fixed in this file instead, so we can safely ignore the warning.
    included_file_warning: ignore
    # treat missing returns as a warning (not a hint)
    missing_return: warning
    # don't treat todos as issues
    todo: ignore
    # don't treat hacks as issues
    hack: ignore
  plugins:
    - dart_code_metrics
  exclude:
    - "**/*.g.dart"
    - "**/gen/*.dart"
    - "**/*.patch.dart"
    - "**/*.pigeon.dart"
    - "**/*.pb.dart"
    - "**/*.pbenum.dart"

# disabled lint rules
linter:
  rules:
    # conflicts with dartfmt
    always_put_control_body_on_new_line: false

    # conflicts with omit_local_variable_types
    always_specify_types: false

    # relative imports are fine
    always_use_package_imports: false

    # no checked exceptions to ensure we're catching every possible error
    avoid_catches_without_on_clauses: false

    # Common pattern for grouping utility functions
    avoid_classes_with_only_static_members: false

    # for easier manual deserialization
    avoid_dynamic_calls: false

    # in practice, only used for variables that go together
    avoid_multiple_declarations_per_line: false

    # can help make code easier to read
    avoid_redundant_argument_values: false

    # sync io are sometimes faster, but block Flutter ui painting causing jank
    avoid_slow_async_io: false

    # dartfmt exposes the dangling else problem
    curly_braces_in_flow_control_structures: false

    # too tedious for minor debugging improvements
    diagnostic_describe_all_properties: false

    # it can be useful to call an asynchronous function without awaiting
    discarded_futures: false

    # this is the recommended way to include variables defined with `flutter build --dart-define`
    do_not_use_environment: false

    # prevents Flutter State classes from being private
    library_private_types_in_public_api: false

    # too tedious for minor readability improvements
    lines_longer_than_80_chars: false

    # feels non-idiomatic
    no_default_cases: false

    # this pattern is often used to avoid name conflicts with state variables
    no_leading_underscores_for_local_identifiers: false

    # conflicts with prefer_single_quotes
    prefer_double_quotes: false

    # doesn't add any value
    prefer_expression_function_bodies: false

    # cumbersome syntax that doesn't help prevent real issues
    prefer_final_parameters: false

    # conflicts with Flutter's WidgetsBindingObserver
    prefer_mixin: false

    # can make it difficult to read import file paths
    prefer_relative_imports: false

    # we should probably do this but it's a lot of work
    public_member_api_docs: false

    # conflicts with prefer_final_locals
    unnecessary_final: false

dart_code_metrics:
  rules:
    - always-remove-listener
    - avoid-collection-methods-with-unrelated-types
    - avoid-duplicate-exports
    - avoid-shrink-wrap-in-lists
    - avoid-unnecessary-setstate
    - avoid-unnecessary-type-assertions
    - avoid-wrapping-in-padding
    - no-boolean-literal-compare
    - no-equal-then-else
    - prefer-correct-edge-insets-constructor
    - prefer-correct-type-name
    - prefer-enums-by-name
    - prefer-immediate-return
    - provide-correct-intl-args

```

</details>

**What did you do? Please include the source code example causing the issue.**

The example of code, executed on CI:

```sh
flutter pub run dart_code_metrics:metrics check-unused-code lib test --fatal-unused
```
```sh
flutter pub run dart_code_metrics:metrics check-unused-files lib test packages/*/lib packages/*/test --fatal-unused --exclude='lib/**/*.pb*.dart'
```

**What did you expect to happen?**

Check should be completed either successfully, or with the report on found issues

**What actually happened?**

Execution failed with
```sh
Run flutter pub run dart_code_metrics:metrics check-unused-code lib test --fatal-unused
pub finished with exit code -9
Error: Process completed with exit code 247.
```

**Are you willing to submit a pull request to fix this bug?**

If the reason will be clear & fix would take reasonable time, then yes, why not

**Comment by incendial** [comment]
> Also, is there any way to collect more data on CI? I haven't found any flag like --verbose in docs, bit it would be handy in this case.

No, unfortunately there is no such option for now.

> pub finished with exit code -9
Error: Process completed with exit code 247.

That's something new, never seen such issues before. Investigating...

**Comment by lavinov-mercury** [comment]
@incendial The issue has gone after upgrading to the latest version. Thank you!

**Comment by incendial** [comment]
> @incendial The issue has gone after upgrading to the latest version. Thank you!

Yay! So happy to hear that!
